src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -114,42 +127,54 @@ private static void AddChannelBindingToken(X509Certificate2 certificate, IntPtr
-                    // revisited if we ever decide to a) introduce a "disallowed" store
-                    // that enables users to "untrust" certs the system trusts, or b) decide
-                    // CRL checking is required, neither of which is done by OpenSSL).
-                    int sslResult = Interop.Crypto.X509VerifyCert(storeCtx);
-                    if (sslResult == 1)
+                    // We need to respect the user's server validation callback if there is one.  If there isn't one,
+                    // we can start by first trying to use OpenSSL's verification, though only if CRL checking is disabled,
+                    // as OpenSSL doesn't do that.
+                    if (easy._handler.ServerCertificateValidationCallback == null &&
+                        !easy._handler.CheckCertificateRevocationList)
                     {
-                        return 1;
-                    }
+                        // Start by using the default verification provided directly by OpenSSL.
+                        // If it succeeds in verifying the cert chain, we're done. Employing this instead of 
+                        // our custom implementation will need to be revisited if we ever decide to introduce a 
+                        // "disallowed" store that enables users to "untrust" certs the system trusts.
+                        int sslResult = Interop.Crypto.X509VerifyCert(storeCtx);
+                        if (sslResult == 1)
+                        {
+                            return 1;
+                        }
 
-                    // X509_verify_cert can return < 0 in the case of programmer error
-                    Debug.Assert(sslResult == 0, "Unexpected error from X509_verify_cert: " + sslResult);
+                        // X509_verify_cert can return < 0 in the case of programmer error
+                        Debug.Assert(sslResult == 0, "Unexpected error from X509_verify_cert: " + sslResult);
+                    }
 
-                    // Only if the fast default verification fails do we then fall back to our more 
-                    // manual and more expensive verification that includes checking the user's 
-                    // certs and not just the system store ones.
+                    // Either OpenSSL verification failed, or there was a server validation callback.
+                    // Either way, fall back to manual and more expensive verification that includes 
+                    // checking the user's certs (not just the system store ones as OpenSSL does).
                     List<X509Certificate2> otherCerts;
                     bool success;
                     using (X509Chain chain = new X509Chain())
                     {
-                        chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                        chain.ChainPolicy.RevocationMode = easy._handler.CheckCertificateRevocationList ? X509RevocationMode.Online : X509RevocationMode.NoCheck;
```

Comment:
Ok.  It wasn't clear from the docs where Online meant "downloading new CRLs is allowed" or "download new CRLs is forced", and whether Offline meant "use the cache; if it isn't cached, fail the request" or "allow using the cache rather than forcing new CRLs to be downloaded".